spec/methods_spec.rb

```diff
@@ -0,0 +1,199 @@
+require '../lib/enumerables.rb'
+
+describe Enumerable do
+let(:array) { [1, 2, 3, 4, 5]}
+let(:range) { (0..5) }
+let(:hash)  { { a: 1, b: 2, c: 3} }
+let(:my_proc) { proc { |num| num + 1} }
+
+    describe '#my_each' do
+         it 'return correct array'
+         expect(array.my_each { |num| num > 2}).to eql(array)
+          end
+         it 'check if it is accepting range' do
+          expect(range.my_each { |num| num > 2}).to eql(range)
+         end
+          it 'check if it is accepting hash' do
+         expect(hash.my_each {|num| num}).to eql(hash)
+         end 
+          it 'when block is not given return Enumerator' do
+          expect(array.my_each).to be_an(Enumerator)
+         end
+    end
+
+    describe '#my_each_with_index' do
+          it 'return correct array' do
+          expect(range.my_each_with_index { |num, index| num}).to eql(hash)
+          end
+          it 'check if it is accepting range' do
+         expect(range.my_each_with_index { |num, index| num > 2}).to eql(range)
+          end
+         it 'check if it is accepting hash' do
+        expect(hash.my_each_with_index { |num, index| num}).to eql(hash)
+         end
+          it 'when block is not given return Enumerator' do
+             expect(array.my_each_with_index).to be_an(Enumerator)
+         end
+    end
+
+    describe '#my_select' do
+        it 'when block is not given return Enumerator' do
+         expect(array.my_each_with_index).to be_an(Enumerator)
+         end
+
+        it 'returns to an array' do
+        expect(array.my_select { |num| num > 3}).to eql([4, 5])
+        end
+
+        it 'returns an array into a range' do
+        expect(array.my_select { |num| num < 3}).to eql([0, 1, 2])
+        end
+
+        it 'returns an array into a hash' do
+        expect(array.my_select { |key, value| value < 2}).to eql({ c: 3})
+        end
+    end
+    describe '#my_all' do
+        it 'if all value for an array are true return true' do
+        expect(array.my_all? { |n| n > 2}).to eql FALSE
+        end
+
+        it 'if all value are false for an array return false' do
+        expect(array.my_all? { |n| n > 6}).to eql FALSE
+        end
+
+        it 'return false when one value is false in a range' do
+        expect(hash.my_all? { |n| n > 2 }).to eql FALSE
+        end
```

Comment:
- [ ] **[REQUIRED]** Use of All Capital `FALSE` is deprecated. Please use the small case `false` instead.